models/repo.go

```diff
@@ -2363,3 +2401,28 @@ func (repo *Repository) CreateNewBranch(doer *User, oldBranchName, branchName st
 
 	return nil
 }
+
+// CreateNewBranchFromCommit creates a new repository branch
+func (repo *Repository) CreateNewBranchFromCommit(doer *User, commit, branchName string) (err error) {
```

Comment:
Maybe if possible make it the same function (renaming `oldBranch` to `baseRef` or similar)